api/v1alpha1/types_etcd_test.go

```diff
@@ -125,19 +97,164 @@ var _ = Describe("Etcd", func() {
 				Controller:         pointer.Bool(true),
 				BlockOwnerDeletion: pointer.Bool(true),
 			}
-			Expect(created.GetAsOwnerReference()).To(Equal(expected))
+			Expect(etcd.GetAsOwnerReference()).To(Equal(expected))
 		})
 	})
 
 	Context("GetRoleName", func() {
 		It("should return the role name for the Etcd", func() {
-			Expect(created.GetRoleName()).To(Equal(GroupVersion.Group + ":etcd:foo"))
+			Expect(etcd.GetRoleName()).To(Equal(GroupVersion.Group + ":etcd:foo"))
 		})
 	})
 
 	Context("GetRoleBindingName", func() {
 		It("should return the rolebinding name for the Etcd", func() {
-			Expect(created.GetRoleName()).To(Equal(GroupVersion.Group + ":etcd:foo"))
+			Expect(etcd.GetRoleName()).To(Equal(GroupVersion.Group + ":etcd:foo"))
+		})
+	})
+
+	Context("IsBackupStoreEnabled", func() {
+		Context("when backup is enabled", func() {
+			It("should return true", func() {
+				Expect(etcd.IsBackupStoreEnabled()).To(Equal(true))
+			})
+		})
+		Context("when backup is not enabled", func() {
+			It("should return false", func() {
+				etcd.Spec.Backup = BackupSpec{}
+				Expect(etcd.IsBackupStoreEnabled()).To(Equal(false))
+			})
+		})
+	})
+
+	Context("IsMarkedForDeletion", func() {
+		Context("when deletion timestamp is not set", func() {
+			It("should return false", func() {
+				Expect(etcd.IsMarkedForDeletion()).To(Equal(false))
+			})
+		})
+		Context("when deletion timestamp is set", func() {
+			It("should return true", func() {
+				etcd.DeletionTimestamp = &metav1.Time{Time: time.Now()}
+				Expect(etcd.IsMarkedForDeletion()).To(Equal(true))
+			})
+		})
+	})
+
+	Context("GetSuspendEtcdSpecReconcileAnnotationKey", func() {
+		ignoreReconciliationAnnotation := IgnoreReconciliationAnnotation
+		suspendEtcdSpecReconcileAnnotation := SuspendEtcdSpecReconcileAnnotation
+		Context("when etcd has only annotation druid.gardener.cloud/ignore-reconciliation set", func() {
+			It("should return druid.gardener.cloud/ignore-reconciliation", func() {
+				etcd.Annotations = map[string]string{
+					IgnoreReconciliationAnnotation: "true",
+				}
+				Expect(etcd.GetSuspendEtcdSpecReconcileAnnotationKey()).To(Equal(&ignoreReconciliationAnnotation))
+			})
```

Comment:
Annotation strings are part of the API and should not change. What we can instead do is instead of having the annotation name as-is one can just have a pointer to it by saying `when etcd has only ignore-reconcile annotation`

Reply:
Addressed in https://github.com/gardener/etcd-druid/pull/777/commits/c0895d233daee0e4aba48a2740cba1b90161adfd